driver/controller.go

```diff
@@ -605,8 +607,54 @@ func (d *controllerService) ListSnapshots(ctx context.Context, req *csi.ListSnap
 // ControllerExpandVolume resizes/updates the volume (not supported yet on Exoscale Public API)
 func (d *controllerService) ControllerExpandVolume(ctx context.Context, req *csi.ControllerExpandVolumeRequest) (*csi.ControllerExpandVolumeResponse, error) {
 	klog.V(4).Infof("ControllerExpandVolume")
+	zone, volumeID, err := getExoscaleID(req.GetVolumeId())
+	if err != nil {
+		return nil, err
+	}
+	client := d.client.WithURL(zone)
+
+	volume, err := client.GetBlockStorageVolume(ctx, volumeID)
+	if err != nil {
+		if errors.Is(err, v3.ErrNotFound) {
+			return nil, status.Errorf(codes.NotFound, "volume %s not found", volumeID)
+		}
+
+		return nil, err
+	}
+
+	newSize, err := getNewVolumeSize(req.GetCapacityRange())
+	if err != nil {
+		return nil, status.Errorf(codes.OutOfRange, "invalid capacity range: %v", err)
+	}
+
+	if newSize < volume.Size {
+		return nil, status.Error(codes.InvalidArgument, "new size must be bigger than actual volume size")
+	}
+
+	_, err = client.ResizeBlockStorageVolume(ctx, volumeID, v3.ResizeBlockStorageVolumeRequest{
+		Size: newSize,
+	})
+	if err != nil {
+		return nil, err
+	}
 
-	return nil, status.Error(codes.Unimplemented, "")
+	nodeExpansionRequired := true
+	volumeCapability := req.GetVolumeCapability()
+	if volumeCapability != nil {
+		err := validateVolumeCapability(volumeCapability)
+		if err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, "volumeCapabilities not supported: %s", err)
+		}
+
+		if _, ok := volumeCapability.GetAccessType().(*csi.VolumeCapability_Block); ok {
+			nodeExpansionRequired = false
+		}
+	}
```

Comment:
wouldn't it be better to do these checks before we do the resize call?
As it is now the block storage volume could get resized and then would you could get the error "volumeCapabilities not supported..." after.